bin/nuxt-build

```diff
@@ -40,49 +40,51 @@ if (argv.help) {
   process.exit(0)
 }
 
-const options = loadNuxtConfig(argv)
+(async function build() {
+  const options = await loadNuxtConfig(argv)
 
-// Create production build when calling `nuxt build`
-options.dev = false
+  // Create production build when calling `nuxt build`
+  options.dev = false
 
-// Analyze option
-options.build = options.build || {}
-if (argv.analyze && typeof options.build.analyze !== 'object') {
-  options.build.analyze = true
-}
+  // Analyze option
+  options.build = options.build || {}
+  if (argv.analyze && typeof options.build.analyze !== 'object') {
+    options.build.analyze = true
+  }
 
-// Silence output when using --quiet
-if (argv.quiet) {
-  options.build.quiet = !!argv.quiet
-}
+  // Silence output when using --quiet
+  if (argv.quiet) {
+    options.build.quiet = !!argv.quiet
+  }
 
-const nuxt = new Nuxt(options)
-const builder = new Builder(nuxt)
+  const nuxt = new Nuxt(options)
+  const builder = new Builder(nuxt)
 
-// Setup hooks
-nuxt.hook('error', err => consola.fatal(err))
+  // Setup hooks
+  nuxt.hook('error', err => consola.fatal(err))
 
-// Close function
-const close = () => {
-  // In analyze mode wait for plugin
-  // emitting assets and opening browser
-  if (options.build.analyze === true || typeof options.build.analyze === 'object') {
-    return
-  }
+  // Close function
+  const close = () => {
+    // In analyze mode wait for plugin
+    // emitting assets and opening browser
+    if (options.build.analyze === true || typeof options.build.analyze === 'object') {
+      return
+    }
 
-  process.exit(0)
-}
+    process.exit(0)
+  }
 
-if (options.mode !== 'spa' || argv.generate === false) {
-  // Build only
-  builder
-    .build()
-    .then(close)
-    .catch(err => consola.fatal(err))
-} else {
-  // Build + Generate for static deployment
-  new Generator(nuxt, builder)
-    .generate({ build: true })
-    .then(close)
-    .catch(err => consola.fatal(err))
-}
+  if (options.mode !== 'spa' || argv.generate === false) {
+    // Build only
+    builder
+      .build()
+      .then(close)
+      .catch(err => consola.fatal(err))
+  } else {
+    // Build + Generate for static deployment
+    new Generator(nuxt, builder)
+      .generate({ build: true })
+      .then(close)
+      .catch(err => consola.fatal(err))
+  }
+})()
```

Comment:
The result of this promise is unhandled. We can call it like this:

```js
build()
  .then(() => { process.exit(0) } )
  .catch(err => { consola.fatal(err); process.exit(1) })
```

Reply:
Right! I guess same goes for the other bin scripts?

Reply:
Yep :) Another idea is to create a wrapper util to refactor this logic. Something like `runScript`.

Impl:

```js
async function runScript(fn) {
  try {
   await fn();
   process.exit(0);
  } catch(e) {
consola.fatal(err)
consola.fatal(err)
 }
}
```

Reply:
But there is the top-level async problem again :see_no_evil:

Reply:
Ah, no. We can avoid it with promises as well.